src/components/EventDetailsComponents/commonControls/MultiColumnRoleNameList.tsx

```diff
@@ -0,0 +1,96 @@
+import { FlatList, View, StyleSheet, LogBox, LayoutChangeEvent } from "react-native";
+import RohText from "@components/RohText";
+import { scaleSize } from "@utils/scaleSize";
+import React, { useEffect, useRef, useImperativeHandle } from 'react';
+import { Colors } from "@themes/Styleguide";
+
+type TMultiColumnRoleNameListProps = {
+    data: Array<{role: string, name: string}>,
+    numColumns: number,
+    setItemHeight(height: number): void
+    setContainerHeight(height: number): void
+    setReachedEnd: (distanceFromEnd: number) => void;
+};
+
+export type TMultiColumnRoleNameListRef = {
+  scrollToEnd?: () => void;
+  scrollToTop?: () => void;
+};
+
+const MultiColumnRoleNameList = React.forwardRef<any,TMultiColumnRoleNameListProps> (
+  (props, ref) => {
+    const {
+      data,
+      numColumns,
+      setItemHeight,
+      setContainerHeight,
+      setReachedEnd,
+    } = props;
+
+    const flatlistRef = useRef<FlatList>(null);
+    useEffect(() => {
+      LogBox.ignoreLogs(['VirtualizedLists should never be nested']);
```

Comment:
What is this???

Reply:
It's because the FlatList is inside a ScrollView further up in the navigation. Sometimes, as here, that's what we want to do anyway. I'm sure there are other ways of solving it, of course.

Reply:
You need to improve it

Reply:
@CSeal Improve what? Please provide some more detailed feedback.

Reply:
From our discussions, I believe that @CSeal means that we should do our own specialisation of VirtualizedList (instead of FlatList) and rebuild the parent scrolling. Which is essentially the optimisations I mentioned above. But I suggest we handle these in conjunction with the UX feedback we expect later.

Reply:
yes, sinewave440hz. right you are. I suggested you improve it, maybe later. using VirtualizedList will be a good solution.